MagicSpellKeyboard/KeyboardViewController.swift

```diff
@@ -137,29 +146,60 @@ class KeyboardViewController: UIInputViewController {
         proxy.insertText("\n")
     }
     
+    @IBAction func didTapShift(sender: UIButton) {
+        if !shiftKeyActive {
+            sender.setTitle("SHIFT ⇧", forState: .Normal)
+            shiftKeyActive = true
+        } else {
+            sender.setTitle("shift ⇧", forState: .Normal)
+            shiftKeyActive = false
+        }
+        updateKeyLabels()
+    }
+    
+    
     @IBAction func touchDownFinger(sender: UIButton) {
         let finger = buttonToFinger[sender]!
         fingersPressed.insert(finger)
 
         switch fingersPressed.count {
         case 1:
-            if let letter = LetterMapping.getLetter(fingersPressed) {
-                proxy.insertText(letter)
+            if shiftKeyActive {
```

Comment:
Consider using a function here. This violates (Don't repeat yourself) DRY principle.